test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+			var genesisBlock_view_full_blocks_list = [
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '1465651642158264047',
+					t_type: 0,
+				},
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '3634383815892709956',
+					t_type: 2,
+				},
+			];
+
+			var blockObject = blocksUtilsModule.readDbRows(
+				genesisBlock_view_full_blocks_list
+			);
 
-				describe('and there are transactions in the block', () => {
-					it('should return the block containing transactions');
-				});
-			});
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('6524861224470851795');
+			expect(blockObject[0].generationSignature).to.equal(
+				'0000000000000000000000000000000000000000000000000000000000000000'
+			);
 		});
 	});
 
 	describe('loadBlocksPart', () => {
-		describe('when there is no error when loading block array', () => {
-			it('should be normalized');
-
-			it('should call callback with error = undefined');
-
-			it('should call callback with blocks as result');
+		it('should return error when library.db.blocks.loadBlocksData fails', done => {
+			library.db.blocks.getHeightByLastId = sinonSandbox.stub().resolves([]);
+			library.db.blocks.loadBlocksData = sinonSandbox
+				.stub()
+				.throws(new Error('An error'));
+
+			blocksUtilsModule.loadBlocksPart({}, (err, blocks) => {
+				expect(loggerStub.error.args[0][0]).to.contains('An error');
+				expect(err).to.equal('Blocks#loadBlockData error');
+				expect(blocks).to.be.undefined;
+				done();
+			});
 		});
 
-		describe('when error is defined', () => {
-			it('should call callback with the error object');
-
-			it('should call callback with blocks as result');
+		it('should return block object', done => {
+			library.db.blocks.loadBlocksData = sinonSandbox
+				.stub()
+				.resolves(viewRow_full_blocks_list);
+
+			library.db.blocks.getHeightByLastId = sinonSandbox
+				.stub()
+				.resolves(library.db.blocks.loadBlocksData);
+
+			blocksUtilsModule.loadBlocksPart({}, (err, blocks) => {
+				expect(err).to.be.null;
+				expect(blocks).to.be.an('array');
+				expect(blocks[0]).to.be.an('object');
+				expect(blocks[0].id).to.equal('13068833527549895884');
+				done();
+			});
 		});
 	});
 
 	describe('loadLastBlock', () => {
-		it('should call library.dbSequence.add');
-
-		it('should call library.db.query to load the last block');
-
-		describe('when db query fails', () => {
-			it('should call logger.error with error stack');
-
-			it('should call callback with the Blocks#loadLastBlock error');
+		it('should return error when library.db.blocks.loadLastBlock fails', done => {
+			library.db.blocks.loadLastBlock = sinonSandbox.stub().resolves(null);
+
+			blocksUtilsModule.loadLastBlock((err, block) => {
+				expect(loggerStub.error.args[0][0]).to.contains(
+					"TypeError: Cannot read property 'length' of null"
```

Comment:
Why we expect type error here? Looks like there is something wrong.

Reply:
It's the expected error if the promise was to yield a value of null.